site/src/pages/CreateWorkspacePage/ExternalAuthButton.tsx

```diff
@@ -27,6 +27,7 @@ export const ExternalAuthButton: FC<ExternalAuthButtonProps> = ({
     <>
       <div css={{ display: "flex", alignItems: "center", gap: 8 }}>
         <LoadingButton
+          id={"external-auth-" + auth.id}
```

Comment:
if this is just for testing purposes, then probably `data-testid`.

Reply:
Yeah, IDs are mainly supposed to be used for associating HTML elements with other elements, or associating elements with browser URL state

I want to say that if you need some kind of "ID-like value"/testing hook for your elements, and the value doesn't have any functionality for the end-user, data attributes (like `data-testid`) are the way to go

Reply:
Another concern is that HTML IDs have to be globally unique. If multiple HTML elements have the same ID, that can cause some nasty jank (e.g., clicking a form input, and a different input getting highlighted)

React has a built-in hook for making it easy to have unique IDs, even when you're working with reusable components
https://react.dev/reference/react/useId